cmd/transfer.go

```diff
@@ -0,0 +1,226 @@
+	"github.com/manifoldco/manifold-cli/generated/provisioning/client/operation"
+	pModels "github.com/manifoldco/manifold-cli/generated/provisioning/models"
+)
+
+func init() {
+	createCmd := cli.Command{
+		Name:      "transfer",
+		ArgsUsage: "[owner]",
+		Usage:     "Transfer a resource",
+		Category:  "RESOURCES",
+		Action: middleware.Chain(middleware.LoadDirPrefs, middleware.EnsureSession,
+			middleware.LoadTeamPrefs, transfer),
+		Flags: append(teamFlags, []cli.Flag{
+			resourceFlag(),
+			projectFlag(),
+			skipFlag(),
+			cli.StringFlag{
+				Name:  "owner, o",
+				Usage: "The new owner for the resource. This can either be a team title or user email",
+			},
+		}...),
+	}
+
+	cmds = append(cmds, createCmd)
+}
+
+func transfer(cliCtx *cli.Context) error {
+	ctx := context.Background()
+
+	if err := exactArgsLength(cliCtx, 1); err != nil {
+		return err
+	}
+
+	newOwner, err := requiredArgName(cliCtx, 0, "owner")
+	if err != nil {
+		return err
+	}
+
+	resourceName, err := validateName(cliCtx, "resource")
+	if err != nil {
+		return err
+	}
+
+	userID, userIDErr := loadUserID(ctx)
+	if userIDErr != nil && userIDErr != errUserActionAsTeam {
+		return userIDErr
+	}
+
+	teamID, err := validateTeamID(cliCtx)
+	if err != nil {
+		return err
+	}
+
+	project, err := validateName(cliCtx, "project")
+	if err != nil {
+		return err
+	}
+
+	client, err := api.New(api.Marketplace, api.Identity, api.Provisioning, api.Analytics)
+	if err != nil {
+		return err
+	}
+
+	var resources []*models.Resource
+
+	resources, err = clients.FetchResources(ctx, client.Marketplace, teamID, project)
+
+	if err != nil {
+		return cli.NewExitError(
+			fmt.Sprintf("Failed to fetch the list of provisioned resources: %s", err), -1)
+	}
+
+	if len(resources) == 0 {
+		return cli.NewExitError("No resources found for transfering", -1)
+	}
+
+	projects, err := clients.FetchProjects(ctx, client.Marketplace, teamID)
+	if err != nil {
+		return cli.NewExitError(
+			fmt.Sprintf("Failed to fetch list of projects: %s", err), -1)
+	}
+
+	var resource *models.Resource
+	if resourceName != "" {
+		var err error
+		resource, err = pickResourcesByName(resources, projects, resourceName)
+		if err != nil {
+			return cli.NewExitError(fmt.Sprintf("Failed to fetch resource: %s", err), -1)
+		}
+	} else {
+		idx, _, err := prompts.SelectResource(resources, projects, resourceName)
+		if err != nil {
+			return prompts.HandleSelectError(err, "Could not select Resource")
+		}
+		resource = resources[idx]
+	}
+
+	fmt.Println(newOwner)
+	newOwnerID, err := newOwnerID(ctx, newOwner, client.Identity)
+	fmt.Println("fooooo", err)
+	if err != nil {
+		return err
+	}
+
+	dontWait := cliCtx.Bool("no-wait")
+	if err := transferResource(ctx, client, userID, teamID, resource.ID, *newOwnerID, dontWait); err != nil {
+	}
+
+	return nil
+}
+
+func newOwnerID(ctx context.Context, newOwner string, ic *iClient.Identity) (*manifold.ID, error) {
+	teams, err := clients.FetchTeams(ctx, ic)
```

Comment:
If the newOwner is not a valid email format, we could just fetch a single team. I'm going to create an issue to add a label filter to GET /teams.